### ISSUE TITLE
The output score of tensorrt model not in range (0, 1)

### ISSUE THREAD

**Issue by wwdok** [open-mmlab/mmaction2#552]
Hi, i used `python tools/pytorch2onnx.py ./configs/recognition/tsm/tsm_r50_video_inference_1x1x8_100e_kinetics400_rgb.py /content/tsm_r50_video_1x1x8_100e_kinetics400_rgb_20210115_2classes.pth --shape 1 8 3 256 340 --verify --show` to convert my trained tsm model to onnx format, then i continued converted it to tensorrt format, but its recognition score is some like this : 
```
[('fall_down', 1.1727579), ('smoking', -1.1995307)]
[('fall_down', 1.655934), ('smoking', -1.6735543)]
[('fall_down', 2.6001232), ('smoking', -2.6183808)]
```
as you can see, the scores are out of range (0, 1).
as @irvingzhang0512 mentioned [here](https://github.com/open-mmlab/mmaction2/issues/410#issuecomment-763371115), if it may be the last F.softmax is not exported to onnx model.
Due to the log is too long which exceed the 65536 characters, so i uploaded it in google drive ([link](https://drive.google.com/file/d/1vpsAL8gA8YbxgQU_MbzYvEVujNeuuA_r/view?usp=sharing))

and this is the visualizaiton of my onnx model : 
![tsm_r50_video_1x1x8_100e_kinetics400_rgb_20210115_2classes-1-8-3-256-340 onnx](https://user-images.githubusercontent.com/43233772/105147337-07164f00-5b3c-11eb-8c17-6544d190cc44.png)

Is the abnormal score range really caused by a mistake in the process of converting the onnx format?

**Comment by innerlee** [comment]
The model does not have softmax/sigmoid (they are not part of backbone), so you may need to add them manually.

![image](https://user-images.githubusercontent.com/9464825/105150379-aab52e80-5b3f-11eb-8393-cdfc349f1fab.png)

**Comment by wwdok** [comment]
Great, the casue is finally found ! but i am not familiar with add operation manually by now, i need to spend some time researching it...

**Comment by wwdok** [comment]
I can add [this softmax function](https://github.com/open-mmlab/mmaction2/blob/81450c834af8d766b4992385686729d7570222dc/mmaction/core/evaluation/accuracy.py#L393:5) into the code to post-process the output score of tensorrt model, but i have no idea how to add softmax op inside the onnx model...

```
def softmax(x, dim=1):
    """Compute softmax values for each sets of scores in x."""
    e_x = np.exp(x - np.max(x, axis=dim, keepdims=True))
    return e_x / e_x.sum(axis=dim, keepdims=True)
```

**Comment by SuX97** [comment]
> I have added this softmax function into the code to post-process the output score of tensorrt model to get the expected score, but i have no idea how to add softmax op inside the onnx model...

@wwdok 
You need to add softmax inside your pytorch module, and then export it. Hopefully onnx/tensorrt can convert it without any pain.

An example:
In your reocgnizer2d.py `forward_dummy`, add:
'''
        outs = (self.cls_head(x, num_segs), )
        + outs = F.softmax(outs)
        return outs
'''

**Comment by kartik1395** [comment]
> > I have added this softmax function into the code to post-process the output score of tensorrt model to get the expected score, but i have no idea how to add softmax op inside the onnx model...
> 
> @wwdok
> You need to add softmax inside your pytorch module, and then export it. Hopefully onnx/tensorrt can convert it without any pain.
> 
> An example:
> In your reocgnizer2d.py `forward_dummy`, add:
> 
> ```
>         outs = (self.cls_head(x, num_segs), )
>         + outs = nn.functional.softmax(outs)
>         return outs
> ```

Hi, I was facing the same issue and I tried your fix to implement softmax layer inside the forward_dummy function but it yields an error.

![torch](https://user-images.githubusercontent.com/52433565/108681917-a313eb80-752a-11eb-85e8-edde566220c4.PNG)

**Comment by kartik1395** [comment]
> > > I have added this softmax function into the code to post-process the output score of tensorrt model to get the expected score, but i have no idea how to add softmax op inside the onnx model...
> > 
> > 
> > @wwdok
> > You need to add softmax inside your pytorch module, and then export it. Hopefully onnx/tensorrt can convert it without any pain.
> > An example:
> > In your reocgnizer2d.py `forward_dummy`, add:
> > ```
> >         outs = (self.cls_head(x, num_segs), )
> >         + outs = nn.functional.softmax(outs)
> >         return outs
> > ```
> 
> Hi, I was facing the same issue and I tried your fix to implement softmax layer inside the forward_dummy function but it yields an error.
> 
> ![torch](https://user-images.githubusercontent.com/52433565/108681917-a313eb80-752a-11eb-85e8-edde566220c4.PNG)

I changed the code to 

`outs = (self.cls_head(x, num_segs), )`
`outs = nn.functional.softmax(outs[0])`
`return (outs, )`

The error seems to go and the conversion is also successful.
